backend/app/src/test/kotlin/io/tolgee/api/v2/controllers/v2KeyController/V2KeyInNamespaceControllerTest.kt

```diff
@@ -106,6 +107,35 @@ class V2KeyInNamespaceControllerTest : ProjectAuthControllerTest("/v2/projects/"
     namespaceService.find("", project.id).assert.isNull()
   }
 
+  @ProjectJWTAuthTestMethod
+  @Test
+  fun `throws error when moving key to default ns where a key with same name already exists`() {
+    val keyName = "super_ultra_cool_key"
+    val namespace = "super_ultra_cool_namespace"
+
+    performProjectAuthPost("keys", CreateKeyDto(name = keyName))
+      .andIsCreated
+    performProjectAuthPost("keys", CreateKeyDto(name = keyName, namespace = namespace))
+      .andIsCreated
+
+    val keyId = keyService.get(project.id, keyName, namespace).id
+
+    performProjectAuthPut("keys/${keyId}/complex-update", ComplexEditKeyDto(name = keyName, namespace = ""))
+      .andIsBadRequest
+      .andAssertError
+      .isCustomValidation.hasMessage("key_exists")
+
+    performProjectAuthPut("keys/${keyId}/complex-update", ComplexEditKeyDto(name = keyName, namespace = null))
+      .andIsBadRequest
+      .andAssertError
+      .isCustomValidation.hasMessage("key_exists")
+
+    performProjectAuthPut("keys/${keyId}/complex-update", ComplexEditKeyDto(name = keyName))
```

Comment:
Since ComplexEditKeyDto has null set as default value for namespace, this just tests the same thing as the previous call. It would make sense to create the dto manually using `mapOf("name" to keyName)`, that way, the namespace would be really missing in the request body, so that would test the desired behaviour.